uw-frame-components/portal/main/controllers.js

```diff
@@ -73,17 +73,31 @@ define(['angular','require'], function(angular, require) {
   }]);
 
   /* Header */
-  app.controller('PortalHeaderController', ['$rootScope', '$scope','$location', 'NAMES', 'APP_FLAGS', 'MISC_URLS', function($rootScope, $scope, $location, NAMES, APP_FLAGS, MISC_URLS) {
+  app.controller('PortalHeaderController', ['$rootScope', '$scope','$location', 'NAMES', 'APP_FLAGS', 'MISC_URLS','notificationsService', function($rootScope, $scope, $location, NAMES, APP_FLAGS, MISC_URLS, notificationsService) {
     this.navbarCollapsed = true;
     this.showLogout = true;
     $scope.showSearch = false;
     $scope.showSearchFocus = false;
     $scope.APP_FLAGS = APP_FLAGS;
     $scope.MISC_URLS = MISC_URLS;
-
+    notificationsService.getAllNotifications().then(function(data) {
+      var notifications = data;
+      var priorityNotifications = [];
+      for( var i=0; i < notifications.length; i++ ) {
```

Comment:
Could use filter filter here.

```javascript
var filteredArray = filterFilter(notifications, {priority === true})
if(filteredArray && filteredArray.length > 0) {
 $scope.hasPriorityNotifications = true;
}
```

Or could add a break in the if statement. Reduces the call through the list of notifications.